core/snap/assertions.go

```diff
@@ -41,20 +42,23 @@ func LookupAssertions(proxyURL string) (assertions, storeID string, err error) {
 
 	res, err := http.DefaultClient.Do(req.WithContext(ctx))
 	if err != nil {
-		return "", "", errors.Annotatef(err, "could not contact snap store proxy at %q. If using an air-gapped proxy you must manually provide the assertions file and store ID", noCredsProxyURL)
+		return "", "", errors.Errorf("could not contact snap store proxy at %q. If using an air-gapped proxy you must manually provide the assertions file and store ID: %w", noCredsProxyURL, err)
 	}
 	defer func() { _ = res.Body.Close() }()
 	if res.StatusCode != http.StatusOK {
-		return "", "", errors.Annotatef(err, "could not retrieve assertions from proxy at %q; proxy replied with unexpected HTTP status code %d", noCredsProxyURL, res.StatusCode)
+		if err != nil {
```

Comment:
This is wrong and must return an error when the status code is not `200 OK`.

Reply:
Thanks and fixed